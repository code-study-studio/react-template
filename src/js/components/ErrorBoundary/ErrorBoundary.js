// @flow

import React, {Component } from 'react';
import type { ChildrenArray } from 'react';

type State = {
  hasError?: boolean,
  children?: ChildrenArray<*>,
};
type Props = {
  hasError?: boolean,
  children?: ChildrenArray<*>,
};

class ErrorBoundary extends Component<State, Props> {
  constructor(props: Props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch(error: string|Error) {
    console.error(error);
    this.setState({ hasError: true });
  }

  render() {
    if (this.state.hasError) {
      return <div className='main-error'>Error: Something went wrong.</div>;
    }
    return this.props.children;
  }
}
export default ErrorBoundary;