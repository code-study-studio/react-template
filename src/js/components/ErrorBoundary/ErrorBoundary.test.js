import React from 'react';
import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary', () => {
  it('should be defined', () => {
    expect(ErrorBoundary).toBeDefined();
  });
  it('should render children', () => {
    const wrapper = shallow(
      <ErrorBoundary><div>children</div></ErrorBoundary>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render children if error', () => {
    const wrapper = shallow(
    <ErrorBoundary><div>children{() => {throw new Error('error');}}</div></ErrorBoundary>);
    expect(wrapper).toMatchSnapshot();
  });
});