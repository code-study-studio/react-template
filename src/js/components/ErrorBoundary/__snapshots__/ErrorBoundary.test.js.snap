// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render children 1`] = `
<div>
  children
</div>
`;

exports[`ErrorBoundary should render children if error 1`] = `
<div>
  children
  <Component />
</div>
`;
